internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -25,32 +25,73 @@ const (
 
 const (
 	annotationCPULoadBalancing = "cpu-load-balancing.crio.io"
+	annotationCPUQuota         = "cpu-quota.crio.io"
+	annotationIRQLoadBalancing = "irq-load-balancing.crio.io"
 	schedDomainDir             = "/proc/sys/kernel/sched_domain"
+	irqSmpAffinityFile         = "/proc/irq/default_smp_affinity"
 )
 
 // HighPerformanceHooks used to run additional hooks that will configure a system for the latency sensitive workloads
 type HighPerformanceHooks struct{}
 
 func (h *HighPerformanceHooks) PreStart(ctx context.Context, c *oci.Container, s *sandbox.Sandbox) error {
 	log.Infof(ctx, "Run %q runtime handler pre-start hook for the container %q", HighPerformance, c.ID())
+
+	if strings.Contains(s.CgroupParent(), "burstable") || strings.Contains(s.CgroupParent(), "besteffort") {
+		log.Infof(ctx, "Container %q is not a guaranteed pod. Skip PreStart.", c.ID())
+		// not guranteed pod, do nothing
+		return nil
+	}
+	if *(c.Spec().Linux.Resources.CPU.Shares)%1024 != 0 {
+		log.Infof(ctx, "Container %q requests cpu share %d: not whole cpu(s). Skip PreStart", c.ID(), *(c.Spec().Linux.Resources.CPU.Shares))
+		// requested cpu not whole number, do nothing
+		return nil
+	}
```

Comment:
there are 3 cases we try to avoid
1. best effort container
2. burstable container
3. guaranteed but request/limit cpu is not whole number of cpu. for example, when request/limit cpu is 0.5, the cpuset will be all cpus in the default cpu pool, so we can't use that cpuset; but cpu.share will be 512. 

cpu.share%1024 == 0 is an easy way to check against case 3.